### PR TITLE
Move things out of helpDB, add some xrefs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by JeffBezanson — APPROVED** [review]

File: base/show.jl [hunk 1]
```diff
@@ -1757,6 +1762,18 @@ function showarray(io::IO, X::AbstractArray, repr::Bool = true; header = true)
     end
 end
 
+"""
+    showcompact(io::IO=STDOUT, x)
```
Comment (tkelman) [inline base/show.jl]: list as two separate signatures, since optional argument syntax only works for trailing arguments

Comment (kshyatt) [inline base/show.jl]: Done, and `make docs` passed locally.

File: base/error.jl [hunk 2]
```diff
@@ -78,7 +78,29 @@ systemerror(p, b::Bool; extrainfo=nothing) = b ? throw(Main.Base.SystemError(str
 
 ## assertion functions and macros ##
 
+
+"""
+    assert(cond)
+
+Throw an [`AssertionError`](@ref) if `cond` is `false`.
+Also available as the macro `@assert expr`.
```
Comment (fredrikekre) [inline base/error.jl]: While you are at it, perhaps change to ``Also available as the macro [`@assert`](@ref).``?

Comment (kshyatt) [inline base/error.jl]: I'm kind of busy rn, if you want to edit the PR to change this (it's a good idea) go wild :)

Comment (kshyatt) [inline base/error.jl]: Nevermind! Got it, and `make docs` passes locally.

**Review by fredrikekre — APPROVED** [review]

**Pull request merged**